### PR TITLE
feat: use hooks for list and display

### PR DESCRIPTION
**Pull request by mdonnalley**
### What does this PR do?

Uses new hook interface from [sf-plugins-core](https://github.com/salesforcecli/sf-plugins-core) for `env list` and `env display`

Notable Changes:
- JSON Schema is now much less specific than it was before. I think this is unavoidable given the federated plugin model
- Removed `--extended` flag from `env list` since we agreed that json output and table output will always have the same information

### What issues does this PR fix or reference?
@W-9811480@
@W-9769891@

### CODE REVIEW

File: src/hooks/envList.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    nonScratchOrgs: [] as OrgAuthorization[],
+    scratchOrgs: [] as OrgAuthorization[],
+  };
+  for (const org of orgs) {
+    if (org.isScratchOrg) {
+      grouped.scratchOrgs = grouped.scratchOrgs.concat(org);
+    } else {
+      grouped.nonScratchOrgs = grouped.nonScratchOrgs.concat(org);
+    }
+  }
```
Comment (peternhale) [inline src/hooks/envList.ts:69]:
``` suggestion
  [scratchOrgs, nonScratchOrgs] = orgs.reduce(([s,n], o) => o.isScratchOrg() ? [[...s, o], n] : [s, [...n, o]], [[],[]]

  const grouped = {
    nonScratchOrgs: nonScratchOrgs as OrgAuthorization[],
    scratchOrgs: scratchOrgs as OrgAuthorization[],
  };
```

**Review by peternhale — APPROVED** [review]
@mdonnalley just one suggestion to replace if/else with lambda

**Review by peternhale — APPROVED** [review]

File: src/hooks/envList.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  const grouped = {
+    nonScratchOrgs: [] as OrgAuthorization[],
+    scratchOrgs: [] as OrgAuthorization[],
+  };
+  for (const org of orgs) {
+    if (org.isScratchOrg) {
+      grouped.scratchOrgs = grouped.scratchOrgs.concat(org);
+    } else {
+      grouped.nonScratchOrgs = grouped.nonScratchOrgs.concat(org);
```
Comment (RodEsp) [inline src/hooks/envList.ts:60]: Why use concat and create new arrays instead of just `.push()`?

**Review by RodEsp — DISMISSED** [review]

**Comment by RodEsp** [comment]
Heya @mdonnalley I'm QA'ing this now and noticed `sf env list` hangs at the end now. I assume this is to do with the hooks, is there anything that could be done about that?

sfEnvListHookHanging

**Review by peternhale — APPROVED** [review]

**Comment by mdonnalley** [comment]
> Heya @mdonnalley I'm QA'ing this now and noticed `sf env list` hangs at the end. I assume this is to do with the hooks, is there anything that could be done about that?
> 
> sfEnvListHookHanging

This will be resolved by https://github.com/oclif/core/pull/243

**Comment by RodEsp** [comment]
> > Heya @mdonnalley I'm QA'ing this now and noticed `sf env list` hangs at the end. I assume this is to do with the hooks, is there anything that could be done about that?
> > ...
> 
> This will be resolved by [oclif/core#243](https://github.com/oclif/core/pull/243)

I'm still seeing the same delay after pulling and building the latest changes and running `sf env list` (even without my test plugin).

**Review by peternhale — APPROVED** [review]

**Pull request merged**